crates/noirc_evaluator/src/ssa_refactor/acir_gen/mod.rs

```diff
@@ -55,26 +56,43 @@ struct Context {
 
     /// Manages and builds the `AcirVar`s to which the converted SSA values refer.
     acir_context: AcirContext,
+
+    initialized_arrays: HashSet<BlockId>,
+}
+
```

Comment:
can you add a doc comment for this? 

Is it only used for dynamic arrays? For arrays with indices known at compile time?

Reply:
Done.
Yes it is only used for arrays with witness indices. Arrays with indices known at compile time will not generate MemoryOp opcodes so this set is not useful for them.